variant-inc/actions-collection
# Feature/cloud 1389 sonar project and scan

# Description

Code changes to create sonar project in sonarcloud if it is not available.

Fixes [#24](https://usxtech.atlassian.net/browse/CLOUD-1389)

## Type of change


## How Has This Been Tested?

```yaml
Test Case 1 [Test on dotnet projects]:
- Steps
  - Delete sonar project in sonar cloud ui.
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-dotnet branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - sample run https://github.com/variant-inc/demo-app/actions/runs/1730483138
 
Test Case 2 [Test on nodejs projects]:
- Steps
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-nodejs branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - Sample run https://github.com/variant-inc/demo-nodejs-app/actions/runs/1730788560

Test Case 3 [Test on python projects]:
- Steps
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-python branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - sample run https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/1741915207
```


## Checklist
